server/webapp/WEB-INF/rails/webpack/views/pages/pac/actions.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ApiResult} from "helpers/api_request_builder";
+import {MithrilViewComponent} from "jsx/mithril-component";
+import m from "mithril";
+import Stream from "mithril/stream";
+import {ConfigReposCRUD} from "models/config_repos/config_repos_crud";
+import {ConfigRepo} from "models/config_repos/types";
+import {Material} from "models/materials/types";
+import * as Buttons from "views/components/buttons";
+import css from "./styles.scss";
+
+interface Attrs {
+  material: Stream<Material>;
+  pluginId: Stream<string>;
+  loc?: LocationHandler;
+}
+
+export interface LocationHandler {
```

Comment:
As this is duplicated from `views/pages/pipelines/actions.tsx`, might be good to extract `LocationHandler` and `WindowLocation` into `helpers/`. Mostly an abstraction used in testing.